prime-router/src/test/kotlin/azure/ReportFunctionTests.kt

```diff
@@ -1201,4 +1201,33 @@ class ReportFunctionTests {
         )
         assert(receiverReturned!!.name == receiver.name)
     }
+
+    @Test
+    fun `return ack if requested`() {
```

Comment:
Can we add a scenario for when it errors out while trying to process an ack message? IE it logs that but then tries to process it as normal. 

Thinking this through, If they send an ack message with a processable message, that gets sent through the pipeline and processed like a normal report. I imagine they would just send the MSH first so it shouldn't get routed anywhere or fail in processing. But maybe in that scenario the endpoint should send a 5xx instead of continuing to process the message.